src/main/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/ops/ExecOpSymmetricHashJoin.java

```diff
@@ -1,22 +1,24 @@
 package se.liu.ida.hefquin.engine.queryplan.executable.impl.ops;
 
+import org.apache.jena.graph.Node;
 import org.apache.jena.sparql.core.Var;
 import se.liu.ida.hefquin.engine.data.SolutionMapping;
 import se.liu.ida.hefquin.engine.data.impl.SolutionMappingUtils;
 import se.liu.ida.hefquin.engine.datastructures.impl.SolutionMappingsHashTable;
 import se.liu.ida.hefquin.engine.datastructures.impl.SolutionMappingsHashTableBasedOnOneVar;
+import se.liu.ida.hefquin.engine.datastructures.impl.SolutionMappingsHashTableBasedOnTwoVars;
+import se.liu.ida.hefquin.engine.datastructures.impl.SolutionMappingsIndexBase;
 import se.liu.ida.hefquin.engine.queryplan.ExpectedVariables;
 import se.liu.ida.hefquin.engine.queryplan.executable.IntermediateResultBlock;
 import se.liu.ida.hefquin.engine.queryplan.executable.IntermediateResultElementSink;
 import se.liu.ida.hefquin.engine.queryproc.ExecutionContext;
 
-import java.util.HashSet;
-import java.util.Set;
+import java.util.*;
 
 public class ExecOpSymmetricHashJoin implements BinaryExecutableOp{
 
-    protected final SolutionMappingsHashTable solMHashTableL;
-    protected final SolutionMappingsHashTable solMHashTableR;
+    protected final SolutionMappingsIndexBase solMHashTableL;
+    protected final SolutionMappingsIndexBase solMHashTableR;
```

Comment:
Is it really necessary that these two are of type  SolutionMappingsIndexBase? Can't they just be of type SolutionMappingsIndex?

Reply:
Yes, we can declare these two as type SolutionMappingsIndex.
Currently, there is no other implementation besides SolutionMappingsIndexBase, and no distinct method is contained in SolutionMappingsIndexBase. Exactly, it is not necessary to declare them as SolutionMappingsIndexBase.